packages/gradient_boosting_model/tests/test_pipeline.py

```diff
@@ -3,41 +3,6 @@
 from gradient_boosting_model.processing.validation import validate_inputs
 
 
-def test_pipeline_drops_unnecessary_features(pipeline_inputs):
-    # Given
-    X_train, X_test, y_train, y_test = pipeline_inputs
-    assert config.model_config.drop_features in X_train.columns
-
-    # When
-    # We use the scikit-learn Pipeline private method `_fit` which is called
-    # by the `fit` method, since this allows us to access the transformed
-    # dataframe. For other models we could use the `transform` method, but
-    # the GradientBoostingRegressor does not have a `transform` method.
-    X_transformed, _ = pipeline.price_pipe._fit(X_train, y_train)
```

Comment:
@solegalli previously I was able to use the `_fit` method to access the transformed dataframe of the pipeline. This method seems to have been updated (possibly here: https://github.com/scikit-learn/scikit-learn/commit/88ce8cd77e7aae1726364d92690025f0027c844c) and is throwing an error. 

Any idea how to access the transformed dataframe for the `GradientBoostingRegressor` (since it doesn't have a `transform` method?)

Reply:
Not straightforward. Here some potential solutions:

https://stackoverflow.com/questions/54332654/how-to-analyse-the-intermediate-steps-of-sklearn-pipeline
https://stackoverflow.com/questions/48743032/get-intermediate-data-state-in-scikit-learn-pipeline?

In short, I think you need to apply all the transformations one after the other, just up to the one before the model.

Reply:
Ok, I asked the sklearn community through the email list and one of the developers told me to do this: 

Using slicing: model[:-1].transform(X)

Reply:
That worked perfectly :)

```
from gradient_boosting_model import pipeline
from gradient_boosting_model.config.core import config

from functools import reduce


def test_pipeline_drops_unnecessary_features(pipeline_inputs):
    # Given
    X_train, X_test, y_train, y_test = pipeline_inputs
    assert config.model_config.drop_features in X_train.columns
    pipeline.price_pipe.fit(X_train, y_train)

    # When
    # We access the transformed inputs with slicing
    transformed_inputs = pipeline.price_pipe[:-1].transform(X_train)

    # Then
    assert config.model_config.drop_features in X_train.columns
    assert config.model_config.drop_features not in transformed_inputs.columns


def test_pipeline_transforms_temporal_features(pipeline_inputs):
    # Given
    X_train, X_test, y_train, y_test = pipeline_inputs

    # When
    # We access the transformed inputs with slicing
    transformed_inputs = pipeline.price_pipe[:-1].transform(X_train)

    # Then
    assert (
        transformed_inputs.iloc[0]["YearRemodAdd"]
        == X_train.iloc[0]["YrSold"] - X_train.iloc[0]["YearRemodAdd"]
    )
```